base/bucket_gocb.go

```diff
@@ -1651,10 +1662,31 @@ func (bucket *CouchbaseBucketGoCB) WriteUpdateWithXattr(k string, xattrKey strin
 				value = nil
 				xattrValue = nil
 			}
+
+			// If user xattrs is enabled
+			if userXattrKey != "" {
+				userXattrCas, err := bucket.GetXattr(k, userXattrKey, &userXattrValue)
```

Comment:
It looks like if you wanted to force a CAS error here in testing, you could add a `GetXattrCallback` to LeakyBucket which goes and touches the doc before GetXattr is actually called, similar to the `WriteWithXattrCallback`.

Reply:
Not possible as its calling directly on CouchbaseBucket rather than Bucket interface